### ISSUE TITLE
jdk_security1_1 FAILED SecurityException: FIPS mode: User-specified 'jdk.certpath.disabledAlgorithms' cannot override profile definition.

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#20320]
**Failure link**
---
From internal [Test_openjdk17_j9_sanity.openjdk_x86-64_windows_fips140_3_openjceplusfips_testList_2](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk17_j9_sanity.openjdk_x86-64_windows_fips140_3_openjceplusfips_testList_2/2/) (win19x86-svl-rt4-1)

```**Java Version**
openjdk version "17.0.13" 2024-10-15
IBM Semeru Runtime Open Edition 17.0.13.0-m1 (build 17.0.13+10)
Eclipse OpenJ9 VM 17.0.13.0-m1 (build v0.48.0-release-8899b66789, JRE 17 Windows Server 2019 amd64-64-Bit Compressed References 20241015_811 (JIT enabled, AOT enabled)
OpenJ9   - 8899b66789
OMR      - f8f0d789a
JCL      - 8e2f05376ed based on jdk-17.0.13+10)
```

[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=customized&amp;TARGET=jdk_security1_1&amp;BASE_DOCKER_REGISTRY_CREDENTIAL_ID=9f50c848-8764-440d-b95a-1d295c21713e&amp;TEST_FLAG=FIPS140_3_OpenJCEPlusFIPS&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk17_j9_sanity.openjdk_x86-64_windows_fips140_3_openjceplusfips&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;separator-f6d5c789-dd25-4194-b271-9092c7ac7f8d=&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_windows&amp;GENERATE_JOBS=true&amp;KEEP_REPORTDIR=true&amp;PERSONAL_BUILD=false&amp;DOCKER_REGISTRY_DIR=&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;RERUN_ITERATIONS=0&amp;SETUP_JCK_RUN=false&amp;DOCKER_REGISTRY_URL_CREDENTIAL_ID=7c1c2c28-650f-49e0-afd1-ca6b60479546&amp;LABEL=&amp;JCK_VERSION=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=https%3A%2F%2Fhyc-runtimes-jenkins.swg-devops.com%2Fjob%2Fbuild-scripts%2Fjob%2Fopenjdk17-pipeline%2F1047%2Fartifact%2Ftarget%2Fwindows%2Fx64%2Fopenj9%2F*.zip%2F*zip*%2Fopenj9.zip&amp;BUILD_IDENTIFIER=&amp;JENKINS_KEY=Jenkins+4096+key&amp;ADOPTOPENJDK_BRANCH=v1.0.4-release&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;BUILD_TYPE=release&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&amp;JDK_VERSION=17&amp;DOCKER_REGISTRY_URL=docker-na.artifactory.swg-devops.com%2Fsys-rt-docker-local&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JCK_GIT_BRANCH=&amp;JDK_REPO=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CLOUD_PROVIDER=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=sys-rt-generic-local%2Chyc-runtimes-team-restricted-oracle-sdk-generic-local&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=2&amp;CONFIG_JSON=%5B%0A++++%7B%0A++++++++%22TEST_FLAG%22+%3A+%22NONE%22%2C+%0A++++++++%22PLATFORM_TARGETS%22+%3A+%5B%0A++++++++++++%7B+%22aarch64_linux%22+%3A+%22defaultTestTargets%2Cdev.external%22+%7D%2C%0A++++++++++++%7B+%22aarch64_mac%22+%3A+%22defaultTestTargets%22+%7D%2C%0A++++++++++++%7B+%22ppc64_aix%22+%3A+%22defaultTestTargets%22+%7D%2C%0A++++++++++++%7B+%22ppc64le_linux%22+%3A+%22defaultTestTargets%2Cdev.external%22+%7D%2C%0A++++++++++++%7B+%22x86-64_linux%22+%3A+%22defaultTestTargets%2Cdev.external%22+%7D%2C%0A++++++++++++%7B+%22x86-64_mac%22+%3A+%22defaultTestTargets%22+%7D%2C%0A++++++++++++%7B+%22x86-64_windows%22+%3A+%22defaultTestTargets%22+%7D%0A++++++++%5D%0A++++%7D%2C%0A++++%7B%0A++++++++%22TEST_FLAG%22+%3A+%22FIPS140_2%22%2C+%0A++++++++%22PLATFORM_TARGETS%22+%3A+%5B%0A++++++++++++%7B+%22ppc64le_linux%22+%3A+%22defaultFipsTestTargets%22+%7D%2C%0A++++++++++++%7B+%22x86-64_linux%22+%3A+%22defaultFipsTestTargets%22+%7D%0A++++++++%5D%0A++++%7D%2C%0A++++%7B%0A++++++++%22TEST_FLAG%22+%3A+%22FIPS140_3_OpenJCEPlusFIPS%22%2C%0A++++++++%22PLATFORM_TARGETS%22+%3A+%5B%0A++++++++++++%7B+%22ppc64_aix%22+%3A+%22defaultFipsTestTargets%22+%7D%2C%0A++++++++++++%7B+%22ppc64le_linux%22+%3A+%22defaultFipsTestTargets%22+%7D%2C%0A++++++++++++%7B+%22x86-64_linux%22+%3A+%22defaultFipsTestTargets%22+%7D%2C%0A++++++++++++%7B+%22x86-64_windows%22+%3A+%22defaultFipsTestTargets%22+%7D%0A++++++++%5D%0A++++%7D%2C%0A++++%7B%0A++++++++%22TEST_FLAG%22+%3A+%22FIPS140_3_OpenJCEPlusFIPS.FIPS140-3%22%2C%0A++++++++%22PLATFORM_TARGETS%22+%3A+%5B%0A++++++++++++%7B+%22ppc64_aix%22+%3A+%22defaultFipsTestTargets%22+%7D%2C%0A++++++++++++%7B+%22ppc64le_linux%22+%3A+%22defaultFipsTestTargets%22+%7D%2C%0A++++++++++++%7B+%22x86-64_linux%22+%3A+%22defaultFipsTestTargets%22+%7D%2C%0A++++++++++++%7B+%22x86-64_windows%22+%3A+%22defaultFipsTestTargets%22+%7D%0A++++++++%5D%0A++++%7D%0A%5D%0A&amp;DOCKERIMAGE_TAG=&amp;TEST_TIME=&amp;JDK_IMPL=openj9&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=%23rt-jenkins&amp;DYNAMIC_COMPILE=false&amp;RELATED_NODES=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;APPLICATION_OPTIONS=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=f3b78d91-4db0-4a87-b4d8-32164430cfdc&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=&amp;TRSS_URL=https%3A%2F%2Ftrssrtp1.fyre.ibm.com%2F&amp;OPENJDK_SHA=&amp;USE_TESTENV_PROPERTIES=true&amp;BUILD_LIST=openjdk&amp;ADDITIONAL_ARTIFACTS_REQUIRED=&amp;UPSTREAM_JOB_NUMBER=&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;separator-87e3a7db-4843-4907-ad26-40f469fe3204=&amp;TIME_LIMIT=36&amp;JVM_OPTIONS=&amp;PARALLEL=None) - Change TARGET to run only the failed test targets

**Optional info**
---
Failure output (captured from console output)
---
```
[2024-10-07T17:31:04.962Z] variation: Mode650
[2024-10-07T17:31:06.189Z] JVM_OPTIONS:  -XX:-UseCompressedOops -Xverbosegclog  -Dsemeru.fips=true -Dsemeru.customprofile=OpenJCEPlusFIPS

[2024-10-07T17:37:04.971Z] TEST: java/security/cert/CertPathBuilder/selfIssued/DisableRevocation.java

[2024-10-07T17:37:04.976Z] STDERR:
[2024-10-07T17:37:04.976Z] java.lang.SecurityException: FIPS mode: User-specified 'jdk.certpath.disabledAlgorithms' cannot override profile definition.
[2024-10-07T17:37:04.976Z] 	at java.base/openj9.internal.security.RestrictedSecurity.checkSetSecurityProperty(RestrictedSecurity.java:441)
[2024-10-07T17:37:04.976Z] 	at java.base/java.security.Security.setProperty(Security.java:793)
[2024-10-07T17:37:04.976Z] 	at DisableRevocation.main(DisableRevocation.java:245)
[2024-10-07T17:37:04.976Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-10-07T17:37:04.976Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-10-07T17:37:04.976Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-10-07T17:37:04.976Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:575)
[2024-10-07T17:37:04.976Z] 	at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
[2024-10-07T17:37:04.976Z] 	at java.base/java.lang.Thread.run(Thread.java:857)

[2024-10-08T13:28:15.163Z] jdk_security1_1_FAILED
```

**Comment by pshipton** [comment]
@jasonkatonica pls take a look. This is from 0.48 M1 testing.

**Comment by jasonkatonica** [comment]
This test is expected to fail when executed in FIPS mode since a user should not be able to set the `jdk.certpath.disabledAlgorithms` as this is blocked when in restricted security mode. 

This test is on the excludes list however we are awaiting the merger of https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/376 such that this gets ignored on Windows.

**Comment by pshipton** [comment]
The exclude list is merged now, and also included in the v0.48.0-release branch.
